File: japiwdbe/WdbeQPrtList.java
/**
  * \file WdbeQPrtList.h
  * Java API code for record of table TblWdbeQPrtList
  * \author Alexander Wirthmueller
  * \date created: 23 Aug 2020
  * \date modified: 23 Aug 2020
  */

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQPrtList {
	
	public WdbeQPrtList(
				int jnum
				, String sref
				, String stubMdlRefWdbeMModule
				, String srefMdlIxVCat
				, String titMdlIxVCat
				, String srefIxVDir
				, String titIxVDir
				, String srefWdbeKHdltype
				, String titSrefWdbeKHdltype
				, int Width
				, String Minmax
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.stubMdlRefWdbeMModule = stubMdlRefWdbeMModule;
		this.srefMdlIxVCat = srefMdlIxVCat;
		this.titMdlIxVCat = titMdlIxVCat;
		this.srefIxVDir = srefIxVDir;
		this.titIxVDir = titIxVDir;
		this.srefWdbeKHdltype = srefWdbeKHdltype;
		this.titSrefWdbeKHdltype = titSrefWdbeKHdltype;
		this.Width = Width;
		this.Minmax = Minmax;
	};

	public int jnum;
	public String sref;
	public String stubMdlRefWdbeMModule;
	public String srefMdlIxVCat;
	public String titMdlIxVCat;
	public String srefIxVDir;
	public String titIxVDir;
	public String srefWdbeKHdltype;
	public String titSrefWdbeKHdltype;
	public int Width;
	public String Minmax;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQPrtList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			stubMdlRefWdbeMModule = Xmlio.extractStringUclc(doc, basexpath, "stubMdlRefWdbeMModule", "mdl", null, 0);
			srefMdlIxVCat = Xmlio.extractStringUclc(doc, basexpath, "srefMdlIxVCat", "mdc", null, 0);
			titMdlIxVCat = Xmlio.extractStringUclc(doc, basexpath, "titMdlIxVCat", "mdc2", null, 0);
			srefIxVDir = Xmlio.extractStringUclc(doc, basexpath, "srefIxVDir", "dir", null, 0);
			titIxVDir = Xmlio.extractStringUclc(doc, basexpath, "titIxVDir", "dir2", null, 0);
			srefWdbeKHdltype = Xmlio.extractStringUclc(doc, basexpath, "srefWdbeKHdltype", "hty", null, 0);
			titSrefWdbeKHdltype = Xmlio.extractStringUclc(doc, basexpath, "titSrefWdbeKHdltype", "hty2", null, 0);
			Width = Xmlio.extractIntegerUclc(doc, basexpath, "Width", "wid", null, 0);
			Minmax = Xmlio.extractStringUclc(doc, basexpath, "Minmax", "mmx", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwdbe/WdbeQPrtList.cpp
/**
	* \file WdbeQPrtList.cpp
	* API code for table TblWdbeQPrtList (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WdbeQPrtList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQPrtList
 ******************************************************************************/

WdbeQPrtList::WdbeQPrtList(
			const uint jnum
			, const string sref
			, const string stubMdlRefWdbeMModule
			, const string srefMdlIxVCat
			, const string titMdlIxVCat
			, const string srefIxVDir
			, const string titIxVDir
			, const string srefWdbeKHdltype
			, const string titSrefWdbeKHdltype
			, const usmallint Width
			, const string Minmax
		) :
			jnum(jnum)
			, sref(sref)
			, stubMdlRefWdbeMModule(stubMdlRefWdbeMModule)
			, srefMdlIxVCat(srefMdlIxVCat)
			, titMdlIxVCat(titMdlIxVCat)
			, srefIxVDir(srefIxVDir)
			, titIxVDir(titIxVDir)
			, srefWdbeKHdltype(srefWdbeKHdltype)
			, titSrefWdbeKHdltype(titSrefWdbeKHdltype)
			, Width(Width)
			, Minmax(Minmax)
		{
};

bool WdbeQPrtList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQPrtList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "stubMdlRefWdbeMModule", "mdl", stubMdlRefWdbeMModule);
		extractStringUclc(docctx, basexpath, "srefMdlIxVCat", "mdc", srefMdlIxVCat);
		extractStringUclc(docctx, basexpath, "titMdlIxVCat", "mdc2", titMdlIxVCat);
		extractStringUclc(docctx, basexpath, "srefIxVDir", "dir", srefIxVDir);
		extractStringUclc(docctx, basexpath, "titIxVDir", "dir2", titIxVDir);
		extractStringUclc(docctx, basexpath, "srefWdbeKHdltype", "hty", srefWdbeKHdltype);
		extractStringUclc(docctx, basexpath, "titSrefWdbeKHdltype", "hty2", titSrefWdbeKHdltype);
		extractUsmallintUclc(docctx, basexpath, "Width", "wid", Width);
		extractStringUclc(docctx, basexpath, "Minmax", "mmx", Minmax);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQPrtList
 ******************************************************************************/

ListWdbeQPrtList::ListWdbeQPrtList() {
};

ListWdbeQPrtList::ListWdbeQPrtList(
			const ListWdbeQPrtList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQPrtList(*(src.nodes[i]));
};

ListWdbeQPrtList& ListWdbeQPrtList::operator=(
			const ListWdbeQPrtList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQPrtList(*(src.nodes[i]));

	return *this;
};

ListWdbeQPrtList::~ListWdbeQPrtList() {
	clear();
};

void ListWdbeQPrtList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQPrtList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQPrtList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQPrtList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQPrtList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQPrtList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQPrtList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};
